src/frontends/wayland/src/main.rs

```diff
@@ -221,12 +302,59 @@ impl KimeContext {
                 self.vk
                     .modifiers(mods_depressed, mods_latched, mods_locked, group);
             }
-            KeyEvent::RepeatInfo { .. } => {
-                // TODO: repeat key
+            KeyEvent::RepeatInfo { rate, delay } => {
+                let info = RepeatInfo { rate, delay };
+                let press_state = self.repeat_state.map(|pair| pair.1);
+                self.repeat_state = Some((info, press_state.unwrap_or(PressState::NotPressing)));
             }
             _ => {}
         }
     }
+
+    pub fn handle_timer_ev(&mut self) -> std::io::Result<()> {
+        // Read timer, this MUST be called or timer will be broken
+        self.timer.read()?;
+
+        if let Some((info, ref mut press_state)) = self.repeat_state {
+            let (key, time) = match press_state {
+                PressState::NotPressing => {
+                    log::warn!("Received timer event while pressing no key.");
+                    return Ok(());
+                }
+                PressState::NotRepeatingYet { key, time } => {
+                    // Start repeat
+                    log::trace!("Start repeating {}", key);
+                    let interval = &Duration::from_secs_f64(1.0 / info.rate as f64);
+                    self.timer.set_timeout_interval(interval)?;
+
+                    let (key, time) = (*key, *time);
+                    *press_state = PressState::Repeating { key, time };
+                    (key, time)
+                }
+                PressState::Repeating { key, time } => {
+                    log::trace!("Keep repeating {}", key);
+                    (*key, *time)
+                }
+            };
+
+            // Emit key repeat event
+            let ev = KeyEvent::Key {
+                serial: self.serial,
+                // NOTE: Not sure if this time should be the time when the key was
+                // initially pressed, or the time of this KeyEvent
+                time,
+                key,
+                // NOTE: KeyState::Repeat is not defined in wayland_client crate.
```

Comment:
There is no `Repeat` KeyState

https://github.com/wayland-project/wayland/blob/ddf21afa6ee99ad6a00705cc74b62a1c4907776b/protocol/wayland.xml#L2240

Reply:
Sorry I got confused due to fcitx5 source code.